src/kid.js

```diff
@@ -15,15 +16,29 @@ export function generateKidMod10(length, baseNum) {
   if (kidLength < 3 || kidLength > 25) {
     throw new Error('Et kid nummer må være mellom 3 og 25 siffer langt.')
   }
-  
   if (baseNum && !Number.isNaN(baseNum)) {
     digits = baseNum.split('').map(num => parseInt(num))
-    return mod10_generate(kidLength, digits)
+    return checkModulo(modulo, { mod10: mod10_generate, mod11: mod11_generate }, kidLength, digits) 
   }
-  
-  return mod10_generate(kidLength - 1)
+
+  return checkModulo(modulo, { mod10: mod10_generate, mod11: mod11_generate }, kidLength - 1)
+}
+
+function checkModulo(modulo, { mod10, mod11 }, ...opts) {
```

Comment:
Isn't the `{ mod10, mod11 }` parameter redundant?
In the switch, you could call `mod10_generate` or `mod11_generate` directly

Also the name `checkModulo` suggests the return type is a boolean, I would rename it to applyModulo or something like that. The doc suggests it can return a boolean, but I don't see where in the function?

Finally, I think the doc for `modulo` is ambigous, as "the modulo" can be interpreted as meaning "the result of a modulo operation". It should be clear in the doc that it is for selecting which modulo function is applied.

Reply:
It is not completely redundant as the mod10/mod11 is either mod10_generate or validate based on who is calling it. Therefore, checkModulo works in both scenarioes here. By calling generate directly in the switch, the validation would stop working and not be as dynamic. I agree with the nameChanges and doc-suggestions. Will fix that

Reply:
checkModulo returns a boolean when mod10/11_validate is used.

Reply:
Can confirm that all functions in the page works as expected. @ergor We should possibly add some unit tests before this project goes out of hand? :D